tests/foreman/cli/test_variables.py

```diff
@@ -0,0 +1,1108 @@
+
+        1. Create a smart Variable with Valid name and default value.
+        2. Attempt to create a variable with same name from same/other class.
+
+        @assert: The variable with same name are not allowed to create from
+        any class.
+
+        @caseautomation: notautomated
+
+        @CaseLevel: System
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_type(self):
+        """Variable is created for variable types - Valid Value.
+
+        Types - string, boolean, integer, real, array, hash, yaml, json
+
+        @id: d89cdd32-dd2b-46fe-bcc2-8c66d92cc6f8
+
+        @steps:
+
+        1.  Create a variable with valid key type and default value.
+
+        @assert: Variable is created with a new type successfully.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_update_type(self):
+        """Variable not created for variable types - Invalid Value.
+
+        Types - string, boolean, integer, real, array, hash, yaml, json
+
+        @id: d64d1b6d-028a-4782-a6d4-e3029d7118b6
+
+        @steps:
+
+        1.  Create a variable with valid key type and invalid default value.
+
+        @assert: Variable is not created with new type for invalid value.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_validate_default_value_required_flag(self):
+        """Error is raised for blank default Value - Required flag.
+
+        @id: cdbe5feb-daf1-49c4-98c0-5bbbd436c3e8
+
+        @steps:
+
+        1.  Create a variable with empty matcher value and required flag to
+        True.
+
+        @assert: Error is raised for blank matcher value.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_validate_default_value_required_flag(self):
+        """Error is not raised for default Value - Required flag.
+
+        @id: 09e42df9-2fe1-444f-a603-6f7b6b3f127e
+
+        @steps:
+
+        1.  Create a variable with some default value and required flag set to
+        True.
+
+        @assert: Error is not raised for default value.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_validate_matcher_value_required_flag(self):
+        """Error is raised for blank matcher Value - Required flag.
+
+        @id: 7e02efb6-e1a2-4872-877a-78075b2de182
+
+        @steps:
+
+        1.  Create a matcher for variable for some attribute with blank value.
+        2.  Set required flag to true.
+
+        @assert: Matcher is not created for blank value.
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_validate_matcher_value_required_flag(self):
+        """Error is not raised for matcher Value - Required flag.
```

Comment:
Add a matcher for some attribute with blank value and when required flag set to true ?